File: algorithm/python/BOJ2164.py
# problem : https://www.acmicpc.net/problem/2164

import sys
from collections import deque

N = int(sys.stdin.readline().rstrip())

deq = deque()

for i in range(1, N + 1):
    deq.append(i)

while len(deq) != 1:
    deq.popleft()
    first = deq.popleft()
    deq.append(first)

print(deq.popleft())

File: algorithm/typescript/BOJ2164.ts
// problem : https://www.acmicpc.net/problem/2164

import fs = require("fs");

const N: number = Number(fs.readFileSync("/dev/stdin").toString());

const queue: number[] = Array.from({ length: N }, (_, index) => index + 1);

while (queue.length !== 1) {
  queue.splice(0, 1);
  const first: number[] = queue.splice(0, 1);
  queue.push(first[0]);
}

console.log(queue[0]);
